server/services/offlinePayment.service.js

```diff
@@ -102,3 +103,112 @@ export const getAllOfflinePayments = async (user, { page = 1, limit = 10 }) => {
   const result = offlinePayments[0].data;
   return { pagination, result };
 };
+
+export const resolveOfflinePayment = async ({ offlinePaymentId, adminId }) => {
+  const offlinePayment = await getOfflinePaymentById(offlinePaymentId).catch((error) => {
+    throw new ErrorHandler(httpStatus.INTERNAL_SERVER_ERROR, 'Internal Server Error', error);
+  });
+  if (!offlinePayment) {
+    throw new ErrorHandler(httpStatus.NOT_FOUND, 'Invalid offline payment');
+  }
+
+  const transaction = {
+    offerId: offlinePayment.offerId,
+    paymentSource: offlinePayment.type,
+    amount: offlinePayment.amount,
+    paidOn: offlinePayment.dateOfPayment,
+    additionalInfo: offlinePayment._id,
+    addedBy: adminId,
+    updatedBy: adminId,
+  };
+
+  try {
+    await addTransaction(transaction).catch((error) => {
+      throw new ErrorHandler(httpStatus.INTERNAL_SERVER_ERROR, 'Internal Server Error', error);
+    });
+
+    return OfflinePayment.findByIdAndUpdate(
+      offlinePayment._id,
+      { $set: { 'resolved.by': adminId, 'resolved.date': Date.now(), 'resolved.status': true } },
+      { new: true },
+    );
+  } catch (error) {
+    throw new ErrorHandler(httpStatus.BAD_REQUEST, 'Error resolving offline payment', error);
+  }
+};
+
+export const makeComment = async ({ comment, user }) => {
+  const offlinePayment = await getOfflinePaymentById(comment.paymentId).catch((error) => {
+    throw new ErrorHandler(httpStatus.INTERNAL_SERVER_ERROR, 'Internal Server Error', error);
+  });
+
+  if (!offlinePayment) {
+    throw new ErrorHandler(httpStatus.NOT_FOUND, 'Invalid offline payment');
+  }
+
+  if (user.role === USER_ROLE.USER && offlinePayment.userId.toString() !== user._id.toString()) {
+    throw new ErrorHandler(httpStatus.FORBIDDEN, 'You are not permitted to perform this action');
+  }
+
+  if (offlinePayment.resolved.status === true) {
```

Comment:
I think I have corrected you on this before 😏 . 

```suggestion
  if (offlinePayment.resolved.status) {
```